src/protocols/core/Subcompositor.cpp

```diff
@@ -99,8 +99,10 @@ CWLSubsurfaceResource::~CWLSubsurfaceResource() {
 }
 
 void CWLSubsurfaceResource::destroy() {
-    if (surface && surface->mapped)
+    if (surface && surface->mapped) {
+        surface->events.unmap.emit();
```

Comment:
pretty sure unmap() will already do it if the surface is mapped.

Reply:
It won't, the callsites need to. We could change that